packages/flutter/lib/src/material/dialog.dart

```diff
@@ -931,15 +931,15 @@ Widget _buildMaterialDialogTransitions(BuildContext context, Animation<double> a
 /// barrier will dismiss the dialog. It is `true` by default and can not be `null`.
 ///
 /// The `barrierColor` argument is used to specify the color of the modal
-/// barrier that darkens everything the dialog. If `null` the default color
+/// barrier that darkens everything below the dialog. If `null` the default color
 /// `Colors.black54` is used.
 ///
 /// The `useSafeArea` argument is used to indicate if the dialog should only
 /// display in 'safe' areas of the screen not used by the operating system
-/// (see [SafeArea] for more details). It is `true` by default which will mean
+/// (see [SafeArea] for more details). It is `true` by default, which means
 /// the dialog will not overlap operating system areas. If it is set to `false`
-/// the dialog will only be constrained by the screen size. It can not be 'null`.
-//
+/// the dialog will only be constrained by the screen size. It can not be `null`.
```

Comment:
nit, it's not your mistake, but would be good to fix this anyway :) 
```suggestion
/// the dialog will only be constrained by the screen size. `useSafeArea` cannot be `null`.
```